src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -381,18 +383,36 @@ object RuleUtils {
       filesAppended: Seq[Path]): LogicalPlan = {
     // Transform the location of LogicalRelation with appended files.
     val planForAppended = originalPlan transformDown {
-      case baseRelation @ LogicalRelation(fsRelation: HadoopFsRelation, baseOutput, _, _) =>
+      case baseRelation @ LogicalRelation(
+            fsRelation @ HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
+            baseOutput,
+            _,
+            _) =>
+        val options = if (location.partitionSchema.isEmpty) {
+          Map[String, String]()
+        } else {
+          // Retrieve basePath from a partition path.
+          val basePath = location.partitionSpec.partitionColumns
+            .foldLeft(location.partitionSpec.partitions.head.path) { (path, _) =>
+              path.getParent
+            }
```

Comment:
the foldLeft is confusing. could you change it as below:
```suggestion
          val basePath: Path = {
            var path = location.partitionSpec.partitions.head.path
            for (_ <- location.partitionSpec.partitionColumns.indices) {
              path = path.getParent
            }
            path
```

Reply:
I prefer FP way - not using `var` if possible.
How about this impl?
```
   val basePath =
            Iterator
              .iterate(location.partitionSpec.partitions.head.path)(_.getParent)
              .drop(location.partitionSpec.partitionColumns.length).next
```

Reply:
+1 for `foldLeft`

Reply:
Ok then let me keep `foldLeft` impl. Thanks!